llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -17395,13 +17395,13 @@ RISCVTargetLowering::getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI,
   return Res;
 }
 
-unsigned
+InlineAsm::ConstraintCode
 RISCVTargetLowering::getInlineAsmMemConstraint(StringRef ConstraintCode) const {
   // Currently only support length 1 constraints.
   if (ConstraintCode.size() == 1) {
     switch (ConstraintCode[0]) {
     case 'A':
-      return InlineAsm::Constraint_A;
+      return InlineAsm::ConstraintCode::A;
```

Comment:
What's up with the swtich statements? :)

Reply:
I don't have much standing for most other backends, but for RISC-V I will put a hard no on refactoring this.

Reply:
+1 Please don't refactor any targets, and especially not in a NFC commit

Reply:
@RKSimon well, this change is refactoring targets in an NFC commit.  I assume you meant "please don't change this switch to an if-else chain" rather than "I reject this PR outright?"

Reply:
Yes - if somebody wants to replace switches with if-else then that should be in a followup patch. But even then, my preference would be to consistently keep to the switch pattern.